tests/tests/test_state_scripts.py

```diff
@@ -348,71 +363,74 @@
             ],
         },
     ),
-    (
-        "Corrupted_script_version_in_etc",
-        {
-            "FailureScript": [],
-            "ExpectedStatus": "failure",
-            "CorruptEtcScriptVersionInUpdate": True,
-            "ScriptOrder": [
-                "Idle_Enter_08_testing",
-                "Idle_Enter_09",
-                "Idle_Leave_09",
-                "Idle_Leave_10",
-                "Sync_Enter_02",
-                "Sync_Enter_03",
-                "Sync_Leave_04",
-                "Sync_Leave_15",
-                "Download_Enter_12",
-                "Download_Enter_13",
-                "Download_Leave_14",
-                "Download_Leave_25",
-                "ArtifactInstall_Enter_01",
-                "ArtifactInstall_Enter_02",
-                "ArtifactInstall_Leave_01",
-                "ArtifactInstall_Leave_03",
-                "ArtifactReboot_Enter_01",
-                "ArtifactReboot_Enter_11",
-                "ArtifactReboot_Leave_01",
-                "ArtifactReboot_Leave_89",
-                "ArtifactReboot_Leave_99",
-                "ArtifactCommit_Enter_01",
-                "ArtifactCommit_Enter_05",
-                "ArtifactCommit_Error_91",
-                "ArtifactRollback_Enter_00",
-                "ArtifactRollback_Enter_01",
-                "ArtifactRollback_Leave_00",
-                "ArtifactRollback_Leave_01",
-                "ArtifactRollbackReboot_Enter_00",
-                "ArtifactRollbackReboot_Enter_99",
-                "ArtifactRollbackReboot_Leave_01",
-                "ArtifactRollbackReboot_Leave_99",
-                "ArtifactFailure_Enter_22",
-                "ArtifactFailure_Enter_33",
-                "ArtifactFailure_Leave_44",
-                "ArtifactFailure_Leave_55",
-            ],
-        },
-    ),
+    # TODO: Move this test case out of the rest, requires full rootfs update
```

Comment:
@kacf What do you think about this test case?

It can't be tested with an update module, as the purpose is to boot into a broken image. We have other tests verifying broken images and the rollback mechanics, so the question here is: how breaking scripts version is different from other kinds of corruptions? Is it worth testing?

Reply:
It is different, because we have no other test which verifies that the client rolls back if the `/etc/mender/scripts/version` file is incorrect. Fortunately, this one is easy to solve: Just add an `ArtifactReboot_Leave` state script which corrupts the version, and an `ArtifactRollbackReboot_Leave` script which restores it.

Reply:
I suspected we had to test this, I wanted to make the skeptical question first :)

Thanks for the good suggestion! I am reworking this test case.